aws/resource_aws_appautoscaling_policy_test.go

```diff
@@ -310,6 +311,44 @@ func TestAccAWSAppautoScalingPolicy_ResourceId_ForceNew(t *testing.T) {
 	})
 }
 
+func TestValidateAppautoscalingPolicyImportInput(t *testing.T) {
```

Comment:
Thanks for adding this unit test; unit test generally appear before acceptance tests. Please move this test to line 14 above `TestAccAWSAppautoScalingPolicy_basic`


In looking at the implementation I wonder if it makes more sense to have `validateAppautoscalingPolicyImportInput` split the input string and validate that each of the parts exists?

```go
serviceNamespace, resourceId, scalableDimension, policyName := validateAppautoscalingPolicyImportInput(test.input)
```

This would mean that you would have to change the implementation [here](https://github.com/terraform-providers/terraform-provider-aws/pull/8397/files#diff-2a2c001f00e78d15d3dad8c4aba0624dR431) but that will also allow for validating the expected number of parts. 


With that being said, are there any changes that need to be to the `TestAccAWSAppautoScalingPolicy_dynamoDb` test to validate your fix?

Reply:
I'll move the test to the "unit test" section. (line 14 above `TestAccAWSAppautoScalingPolicy_basic`).

Yes, I do agree it makes sense to have everything related to the validation inside the function. I'll move that inside `validateAppautoscalingPolicyImportInput`.

Yes, the changes required to `TestAccAWSAppautoScalingPolicy_dynamoDb` are related to the input expected on the [import command](https://www.terraform.io/docs/providers/aws/r/appautoscaling_policy.html#import) because the `ResourceId` and `PolicyName` format varies depending on the resource (ECS, RDS or DynamoDB).

Reply:
- Unit test moved to line 14
- Addded negative unit test cases
- Refactored to have all validations inside `validateAppautoscalingPolicyImportInput`

```bash
make t estacc TEST=./aws TESTARGS='-run=TestValidateAppautoscalingPolicyImportInput_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestValidateAppautoscalingPolicyImportInput_ -timeout 120m
=== RUN   TestValidateAppautoscalingPolicyImportInput_positiveCases
--- PASS: TestValidateAppautoscalingPolicyImportInput_positiveCases (0.00s)
=== RUN   TestValidateAppautoscalingPolicyImportInput_negativeCases
--- PASS: TestValidateAppautoscalingPolicyImportInput_negativeCases (0.00s)
```

I was not able to run the integration tests to reproduce the error. I'll keep trying and let you know

Reply:
Ok, I was able to reproduce and fix the integration test. Below the test results
```bash
make testacc TESTARGS="-run=TestAccAWSAppautoScaling Policy_dynamoDb"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -parallel 20 -run=TestAccAWSAppautoScalingPolicy_dynamoDb -timeout 120m
?       github.com/terraform-providers/terraform-provider-aws   [no test files]
=== RUN   TestAccAWSAppautoScalingPolicy_dynamoDb
=== PAUSE TestAccAWSAppautoScalingPolicy_dynamoDb
=== CONT  TestAccAWSAppautoScalingPolicy_dynamoDb
--- PASS: TestAccAWSAppautoScalingPolicy_dynamoDb (147.54s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       147.597s
```